model/src/main/java/oracle/kubernetes/weblogic/domain/ConfigurationNotSupportedException.java

```diff
@@ -0,0 +1,24 @@
+// Copyright 2018, Oracle Corporation and/or its affiliates.  All rights reserved.
+// Licensed under the Universal Permissive License v 1.0 as shown at
+// http://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.weblogic.domain;
+
+/**
+ * An exception to be thrown on an attempt to configure a field not supported by the domain model.
+ * Only used in unit testing.
+ */
+public class ConfigurationNotSupportedException extends RuntimeException {
+  private final String context;
+  private final String field;
+
+  public ConfigurationNotSupportedException(String context, String field) {
+    this.context = context;
+    this.field = field;
+  }
+
+  @Override
+  public String getMessage() {
+    return String.format("Configuring field %s is not supported for a %s", field, context);
```

Comment:
I18n

Reply:
The configuration classes are intended for unit tests only. As noted in oracle.kubernetes.weblogic.domain.DomainConfigurator:

/**
 * Configures a domain, adding settings independently of the version of the domain representation.
 * Note that the configurator uses a predefined domain schema, and should only be used for testing.
 * Using it in the runtime runs the risk of corrupting the domain.
 */

The code is sitting in "main" rather than "test" because it is needed for testing, not only the model itself, but also the operator runtime and the installation (and ideally) integration tests. Because it depends on the model and is also used by model tests, I see no other place to put it.